### PR TITLE
Limit process access and actions

### PR DESCRIPTION
**Pull request by balsarori**
Ref: #3850

### CODE REVIEW

File: activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeIT.java [hunk 1]
```diff
@@ -894,4 +902,60 @@ public void should_not_returnProcessesToNonInitiatorUser() {
...
+        securityUtil.logInAs("garth");
+
+        //when
+        processInstance = processRuntime.processInstance(processInstance.getId());
+
+        assertThat(processInstance.getInitiator()).isEqualTo("user");
+    }
+
+    @Test
+    public void should_ForbidCancelingProcessInstanceUsingNonInitiatorUser() {
```
Comment (mauriziovitale) [inline activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeIT.java]: if garth involved in one of the tasks or completely external?

Comment (mauriziovitale) [reply activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeIT.java]: should_ForbidCancelingProcessInstanceUsingTaskInvolvedUser ?
```suggestion
    public void should_ForbidCancelingProcessInstanceUsingNonInitiatorUser() {
```

probably is more specific?

**Review by mauriziovitale — COMMENTED** [review]
check the comment

File: activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeIT.java (hunk 1 above)
Comment (balsarori) [reply activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeIT.java]: grath is involved in this case. However, only initiator should be allowed to cancel the process thus the name `should_ForbidCancelingProcessInstanceUsingNonInitiatorUser `

**Review by mauriziovitale — APPROVED** [review]

File: activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ProcessRuntimeImpl.java [hunk 2]
```diff
@@ -491,4 +496,39 @@ public Deployment selectLatestDeployment(){
...
+            securityManager.getAuthenticatedUserId().equals(processInstance.getStartUserId());
+    }
+
+    private boolean isATaskAssigneeOrACandidate(String processInstanceId) {
+        String authenticatedUserId = securityManager.getAuthenticatedUserId();
+        TaskQuery taskQuery = taskService.createTaskQuery().processInstanceId(processInstanceId);
+        taskQuery.or()
+            .taskCandidateOrAssigned(securityManager.getAuthenticatedUserId(),
+                securityManager.getAuthenticatedUserGroups())
+            .taskOwner(authenticatedUserId)
```
Comment (Giovanni007) [inline activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ProcessRuntimeImpl.java:528]: who is the taskOwner in this case?

File: activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeIT.java [hunk 3]
```diff
@@ -894,4 +902,60 @@ public void should_not_returnProcessesToNonInitiatorUser() {
...
+        securityUtil.logInAs("garth");
+
+        //when
+        processInstance = processRuntime.processInstance(processInstance.getId());
+
+        assertThat(processInstance.getInitiator()).isEqualTo("user");
+    }
+
+    @Test
+    public void should_returnProcessToTaskCandidate() {
```
Comment (Giovanni007) [inline activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeIT.java:922]: maybe we could split this is in  :
should_returnProcessToTaskCandidateUser
should_returnProcessToTaskCandidateGroup

**Review by Giovanni007 — APPROVED** [review]
Some minor comment. LGTM!

File: activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ProcessRuntimeImpl.java (hunk 2 above)
Comment (balsarori) [reply activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ProcessRuntimeImpl.java:528]: Whoever is set to be the owner of the task. Setting the owner is only available via the API not possible via process definition.

**Pull request merged**